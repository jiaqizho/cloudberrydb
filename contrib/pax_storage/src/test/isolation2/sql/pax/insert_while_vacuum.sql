-- @Description Ensures that an insert during a vacuum operation is ok
-- 
DROP TABLE IF EXISTS pax_tbl;
CREATE TABLE pax_tbl (a INT);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);

DELETE FROM pax_tbl WHERE a < 128;
2>: VACUUM pax_tbl;
4: BEGIN;
4: insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);COMMIT;
2<:
3: SELECT COUNT(*) FROM pax_tbl WHERE a = 1500;
4: INSERT INTO pax_tbl VALUES (0);
