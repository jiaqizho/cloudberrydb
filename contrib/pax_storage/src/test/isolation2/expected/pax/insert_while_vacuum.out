-- @Description Ensures that an insert during a vacuum operation is ok
--
DROP TABLE IF EXISTS pax_tbl;
DROP
CREATE TABLE pax_tbl (a INT);
CREATE
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000

DELETE FROM pax_tbl WHERE a < 128;
DELETE 2667
2>: VACUUM pax_tbl;  <waiting ...>
4: BEGIN;
BEGIN
4: insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);insert into pax_tbl select generate_series(1001,2000);COMMIT;
INSERT
2<:  <... completed>
VACUUM
3: SELECT COUNT(*) FROM pax_tbl WHERE a = 1500;
 count 
-------
 20    
(1 row)
4: INSERT INTO pax_tbl VALUES (0);
INSERT 1
